Apps/GatewayApi/src/Services/DependentService.cs

```diff
@@ -81,103 +83,56 @@ public class DependentService : IDependentService
-                        ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Patient),
-                    },
-                };
-            }
+                case ResultType.Error:
+                    return RequestResultFactory.ServiceError<DependentModel>(ErrorType.CommunicationExternal, ServiceType.Patient, "Communication Exception when trying to retrieve the Dependent");
 
-            if (patientResult.ResultStatus == ResultType.ActionRequired)
-            {
-                return new RequestResult<DependentModel>
-                {
-                    ResultStatus = ResultType.ActionRequired,
-                    ResultError = ErrorTranslator.ActionRequired(ErrorMessages.DataMismatch, ActionType.DataMismatch),
-                };
-            }
+                case ResultType.ActionRequired:
+                    return RequestResultFactory.ActionRequired<DependentModel>(ActionType.DataMismatch, ErrorMessages.DataMismatch);
 
-            this.logger.LogDebug("Finished getting dependent details... {DependentPhn}", addDependentRequest.Phn);
+                default:
+                    if (!IsMatch(addDependentRequest, patientResult.ResourcePayload))
+                    {
+                        return RequestResultFactory.ActionRequired<DependentModel>(ActionType.DataMismatch, ErrorMessages.DataMismatch);
+                    }
 
-            // Verify dependent's details entered by user
-            if (patientResult.ResourcePayload == null || !this.ValidateDependent(addDependentRequest, patientResult.ResourcePayload))
-            {
-                this.logger.LogDebug("Dependent information does not match request: {DependentPhn}", addDependentRequest.Phn);
-                return new RequestResult<DependentModel>
-                {
-                    ResultStatus = ResultType.ActionRequired,
-                    ResultError = ErrorTranslator.ActionRequired(ErrorMessages.DataMismatch, ActionType.DataMismatch),
-                };
-            }
+                    // Verify dependent has HDID
+                    else if (string.IsNullOrEmpty(patientResult.ResourcePayload.HdId))
+                    {
+                        return RequestResultFactory.ActionRequired<DependentModel>(ActionType.NoHdId, ErrorMessages.InvalidServicesCard);
+                    }
 
-            // Verify dependent has HDID
-            if (string.IsNullOrEmpty(patientResult.ResourcePayload.HdId))
-            {
-                return new RequestResult<DependentModel>
-                {
-                    ResultStatus = ResultType.ActionRequired,
-                    ResultError = ErrorTranslator.ActionRequired(ErrorMessages.InvalidServicesCard, ActionType.NoHdId),
-                };
+                    break;
             }
 
             // Insert Dependent to database
-            ResourceDelegate dependent = new()
+            var dependent = new ResourceDelegate()
```

Comment:
lines 118 and line 126.  Why was this changed?  Before you onboarded there were a lot changes made to avoid warnings.  When I try to make that change you have done on line 126, my Rider IDE will not allow me to do that.

Reply:
I find the C# 9.0 constructor invocation pattern less readable and less flexible when following OO/SOLID principles (https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new for some caveats with it).
I don't use Rider yet so I don't know why it wouldn't let you change the syntax to standard 20 years old C#.

Reply:
If you like look at our code base, that seems to be the standard. I recall a task earlier to fix the code base to ensure new() was being used correctly.  I will approve but you may want to discuss with @sslaws